### PR TITLE
Append ClusterName to AppName for CI Tests

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
This change appends the `clusterName` to any deployed apps when deploying against the BaaS in Atlas.  This was done to ensure the apps are cleaned up properly when the tests are finished.

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: integration-tests/tests/src/utils/import-app.ts [hunk 1]
```diff
@@ -56,8 +56,10 @@ export function getDefaultReplacements(name: string): TemplateReplacements {
   // When running on CI we connect through mongodb-atlas instead of local-mongodb
   const { mongodbClusterName } = environment;
   if (typeof mongodbClusterName === "string") {
+    const replacements = { "config.json": { name: `${name}-${mongodbClusterName}` } };
```
Comment (kraenhansen) [inline integration-tests/tests/src/utils/import-app.ts:59]: Could we move this to a separate `appName` const and reuse that below?
I remember Nikola mentioning that the app name needs to be below 31 symbols - how do we guarantee that?

Comment (kraenhansen) [reply integration-tests/tests/src/utils/import-app.ts:59]: I remember Nikola mentioning that the app name needs to be below 31 symbols - how do we guarantee that?

File: integration-tests/tests/src/utils/import-app.ts [hunk 2]
```diff
@@ -72,6 +74,8 @@ export function getDefaultReplacements(name: string): TemplateReplacements {
       };
     } else if (name === "with-db-flx") {
       return {
+        ...replacements,
```
Comment (kraenhansen) [inline integration-tests/tests/src/utils/import-app.ts:77]: Perhaps we need a bit deeper merge of these objects?

File: integration-tests/tests/src/utils/import-app.ts (hunk 1 above)
Comment (takameyer) [reply integration-tests/tests/src/utils/import-app.ts:59]: We could throw an exception 🤷🏼 That's in our hands to create valid app names.  The clusterName.length should stay consistent in CI.

File: integration-tests/tests/src/utils/import-app.ts (hunk 2 above)
Comment (takameyer) [reply integration-tests/tests/src/utils/import-app.ts:77]: At least in the case of the base `config.json`, the object is flat

File: integration-tests/tests/src/utils/import-app.ts [hunk 3]
```diff
@@ -72,6 +74,8 @@ export function getDefaultReplacements(name: string): TemplateReplacements {
...
+        "config.json": { name: `${name}-${mongodbClusterName}` },
```
Comment (takameyer) [inline integration-tests/tests/src/utils/import-app.ts:78]: I will remove this line

File: packages/realm-app-importer/src/AppImporter.ts [hunk 4]
```diff
@@ -233,9 +233,10 @@ export class AppImporter {
     }
   }
 
-  private loadAppConfigJson(appTemplatePath: string): AppConfig {
+  private loadAppConfigJson(appTemplatePath: string, replacements: TemplateReplacements = {}): AppConfig {
     const configJsonPath = path.resolve(appTemplatePath, "config.json");
-    return this.loadJson(configJsonPath);
+    const configJson = this.loadJson(configJsonPath);
+    return replacements["config.json"] ? { ...configJson, ...replacements["config.json"] } : configJson;
```
Comment (kraenhansen) [inline packages/realm-app-importer/src/AppImporter.ts:239]: This seems simpler.

```suggestion
    return { ...configJson, ...replacements["config.json"] };
```

File: packages/realm-app-importer/src/AppImporter.ts [hunk 5]
```diff
@@ -233,9 +233,10 @@ export class AppImporter {
     }
   }
 
-  private loadAppConfigJson(appTemplatePath: string): AppConfig {
+  private loadAppConfigJson(appTemplatePath: string, replacements: TemplateReplacements = {}): AppConfig {
```
Comment (kraenhansen) [inline packages/realm-app-importer/src/AppImporter.ts:236]: Perhaps this should take the `appName` as a (optional) parameter instead?

File: integration-tests/tests/src/utils/import-app.ts (hunk 2 above)
Comment (kraenhansen) [reply integration-tests/tests/src/utils/import-app.ts]: Never mind - i assumed this was provided as an argument, didn't read all the code above.

File: packages/realm-app-importer/src/AppImporter.ts (hunk 5 above)
Comment (takameyer) [reply packages/realm-app-importer/src/AppImporter.ts:236]: Yeah, we can try that.  The underlying code now looks like this:
```typescript
  private loadAppConfigJson(appTemplatePath: string, appNameOverride?: string): AppConfig {
    const configJsonPath = path.resolve(appTemplatePath, "config.json");
    const configJson = this.loadJson(configJsonPath);
    return appNameOverride ? { ...configJson, ...{ name: appNameOverride } } : configJson;
  }
```

Comment (takameyer) [reply packages/realm-app-importer/src/AppImporter.ts:236]: and now:
```typscript
    const appNameOverride = { ...replacements["config.json"] }?.name as string;
 ```

Comment (takameyer) [reply packages/realm-app-importer/src/AppImporter.ts:236]: Kind of like the other way better, as we are "loading" `AppConfigJson`, which could possibly include other template replacements other than `name`

File: packages/realm-app-importer/src/AppImporter.ts (hunk 4 above)
Comment (takameyer) [reply packages/realm-app-importer/src/AppImporter.ts]: Good point, i'll change it

**Review by kneth — APPROVED** [review]
LGTM

**Pull request merged**